### ISSUE TITLE
[Feature]: Allow custom skype command

### ISSUE THREAD

**Issue by bilderbuchi** [sagetarian/skype-wrapper#2]
Hi!

I'd like to request that this script can start skype with a customizable command. In my case, I got the skype profile stored in a non-default location (to be able to share it across a multi-boot environment), and I start skype with `skype --dbpath=/path/to/profile`, it would be great if this script could also do that.

**Comment by sagetarian** [comment]
cool, i'll add a setting in gsettings tomorrow .. just a bit busy :) .. for now, you can run skype the way you do and then start skype-wrapper afterwards .. it will attach itself to your running skype .. the only issue is that skype will be seperate to the skype-wrapper unity launcher :) the launcher will still only bring focus to your running skype, but you'll have 2 skype icons on your unity launcher ..

its a work around until i can fix this .. will get back to you :)

**Comment by bilderbuchi** [comment]
no workaround or hurry needed. currently, I use the method from http://www.webupd8.org/2011/05/skype-ubuntu-messaging-menu-notifyosd.html and it mostly works ok.

**Comment by sagetarian** [comment]
awesome ;) yeah, i used to use that but if i remember correctly the indicators dont look nice or work for Group Chats .. its the reason why i forked skype-wrapper .. plus now we have avatars in our notifications :D 

what you can do in the meantime is

<code>sudo gedit /usr/share/skype-wrapper/skype-wrapper.py</code>

just before

<code>ret = subprocess.call(['python','indicator-applet-skype.py'])<code>

add

<code>os.system("skype --dbpath=/path/to/profile")<code>

**Comment by sagetarian** [comment]
cool .. i'm done

i'd be happy to ask if you can test if it works :)

its version 0.5.2.5 in the debs section

it will create a setting in gsettings (or dconf-editor) under apps/skype-wrapper called "start-skype-cmd-params"

**Comment by bilderbuchi** [comment]
Thanks! I take it you just tested it successfully yourself? Then I'll wait for the PPA. Will report back if anything goes wrong. :-) (Sorry I just that moment came around to being able to test it)

**Comment by bilderbuchi** [comment]
works like a charm as far as i can see. thanks!

**Comment by bilderbuchi** [comment]
It seems this does not really work, and somehow kills notification functionality. See Issue #17

**Comment by sagetarian** [comment]
#facepalm .. gotta love python-skype .. i'll try to rework this

**Comment by bilderbuchi** [comment]
using 0ubuntu5-4, this still doesn't work - no notifications.

**Comment by sagetarian** [comment]
can you try adding these to that commandline, i'll do it automatically for others, but want to check if it works first

--enable-dbus --use-session-dbus

**Comment by bilderbuchi** [comment]
doesn't change anything. quitting skype still restarts it, btw.

log with loglevel 0:

python-skype: 1.0.31.0-2
python-imaging: 1.1.7-3ubuntu1
Python-2.6, 2.7
WARNING: Checking unread messages failed: Skype command timeout
WARNING: Checking unread messages failed: Skype command timeout
WARNING: Checking online status changing users failed
WARNING: Checking online presence failed
WARNING: Checking file transfers failed (Skype attach timeout)
WARNING: Checking unread messages failed: Skype attach timeout
WARNING: Checking online status changing users failed
WARNING: Checking online status changing users failed
WARNING: Checking online presence failed
WARNING: Skype instance has terminated, exiting
WARNING: Checking unread messages failed: Skype command timeout
WARNING: Checking unread messages failed: Skype command timeout
WARNING: Checking unread messages failed: Skype command timeout
WARNING: Checking unread messages failed: Skype command timeout
WARNING: Checking unread messages failed: Skype command timeout
WARNING: Checking unread messages failed: Skype command timeout
WARNING: Checking unread messages failed: Skype command timeout
WARNING: Checking online status changing users failed
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Initializing Skype API
INFO: Starting Skype
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
WARNING: Failed to attach skype-wrapper to Skype process
INFO: Initializing Skype API
INFO: Starting Skype
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
WARNING: Failed to attach skype-wrapper to Skype process
INFO: Initializing Skype API
INFO: Starting Skype
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
WARNING: Failed to attach skype-wrapper to Skype process
INFO: Initializing Skype API
INFO: Starting Skype
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Initializing Skype API
INFO: Starting Skype
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Initializing Skype API
INFO: Starting Skype
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Initializing Skype API
INFO: Starting Skype

**Comment by bilderbuchi** [comment]
btw, should we track this here, or in #17?

**Comment by sagetarian** [comment]
lol, here might be best until we actually finish this feature..

this sucks .. looks like python-skype can't attach to a skype that has a different dbpath .. 

one more test to make sure, before i push the bug to onto the python-skype people .. 
1. Terminate skype-wrapper and indicator-skype
2. Start skype itself with those dbpaths
3. Run skype-wrapper in terminal and see if she attaches to the skype process and does notifications and all

what i aim to achieve by this test?

check if its my commandline script thats causing an issue, or if python-skype has the limitation

**Comment by bilderbuchi** [comment]
huh, that worked. at least i get online notifications. only thing amiss (in a short test) is that there are 2 skype icons in UMM

log as before:
python-skype: 1.0.31.0-2
python-imaging: 1.1.7-3ubuntu1
Python-2.6, 2.7
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Attached complete
INFO: Checking unread messages
WARNING: Checking unread messages failed: Skype command timeout
INFO: Checking online status changing users
INFO: User anoxaemia is online
INFO: Found avatar for anoxaemia
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-anoxaemia.jpg
INFO: User opampop is online
INFO: Found avatar for opampop
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-opampop.jpg
INFO: User deejaysevero is online
INFO: Found avatar for deejaysevero
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-deejaysevero.jpg
INFO: User desaxismundi is online
INFO: Found avatar for desaxismundi
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-desaxismundi.jpg
INFO: User kiilo_ is online
INFO: Found avatar for kiilo_
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-kiilo_.jpg
INFO: User jvulliet is online
INFO: Found avatar for jvulliet
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-jvulliet.jpg
INFO: User sebastian_gregor is online
INFO: Found avatar for sebastian_gregor
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-sebastian_gregor.jpg
INFO: User giorgio.rinolfi is online
INFO: Found avatar for giorgio.rinolfi
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-giorgio.rinolfi.jpg
INFO: User hubertom80 is online
INFO: Found avatar for hubertom80
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-hubertom80.jpg
INFO: User solanina_ is online
INFO: Found avatar for solanina_
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-solanina_.jpg
INFO: User m.willitsch.atec is online
INFO: Found avatar for m.willitsch.atec
INFO: User schandi80 is online
INFO: Found avatar for schandi80
INFO: Checking online presence
INFO: Checking file transfers
INFO: Check if Skype instance is running
INFO: Checking online presence
INFO: Check if Skype instance is running
INFO: Check if Skype instance is running
INFO: Check if Skype instance is running
INFO: Check if Skype instance is running
INFO: Check if Skype instance is running
(this is continuing in ~500ms intervals...)

**Comment by sagetarian** [comment]
the 500ms intervals = good .. just makes sure that when you quit skype, or terminate it in the terminal then we quit skype-wrapper

try running this .. 
1. what i did was run the "skype COMMANDLINE" which opens the client (as in the previous version, no changes) 
2. quit the indicator-skype process and tell skype-wrapper to respawn the program.

What i plan to achieve by this?

I plan to synthesis you typing in the terminal the commandline and then only run indicator-skype in a fresh run .. this should in theory replicate all conditions..

https://github.com/downloads/sagetarian/skype-wrapper/skype-wrapper_0ubuntu5-4.1_all.deb

just to confirm, you don't run skype in "SUDO" ?

Also this should stop the respawning of skype when you quit skype and its not successfully attached

**Comment by bilderbuchi** [comment]
Sorry, I don't understand, could you rephrase that?
I should
1) kill skype-wrapper and indicator-skype
2) start skype with the dbpath.
3) ??? (indicator-skype is not running at that moment) and neither is skype-wrapper)

Just in case you that's what you want, I did:
1) kill skype-wrapper and indicator-skype
2) start skype with the dbpath.
3) start skype-wrapper. 
4) kill indicator-skype. skype-wrapper process is finished too. see log1
5) start skype-wrapper again. see log 2

both after 4 and 5, notifications worked. 
no, i don't run skype with sudo.
used skype-wrapper 0ubuntu5-4.2~oneiric2

## log1:

INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Attached complete
INFO: Checking unread messages
WARNING: Checking unread messages failed: Skype command timeout
INFO: Checking online status changing users
INFO: User anoxaemia is online
INFO: Found avatar for anoxaemia
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-anoxaemia.jpg
INFO: User olikuschel is online
INFO: Found avatar for olikuschel
INFO: User opampop is online
INFO: Found avatar for opampop
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-opampop.jpg
INFO: User deejaysevero is online
INFO: Found avatar for deejaysevero
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-deejaysevero.jpg
INFO: User tonfilm is online
INFO: Found avatar for tonfilm
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-tonfilm.jpg
INFO: User death.to.goemb is online
INFO: Found avatar for death.to.goemb
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-death.to.goemb.jpg
INFO: User antokhio is online
INFO: Found avatar for antokhio
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-antokhio.jpg
INFO: Checking online presence
INFO: Checking file transfers
INFO: Check if Skype instance is running
INFO: Checking online presence
INFO: Check if Skype instance is running
etc..

log2:
python-skype: 1.0.31.0-2
python-imaging: 1.1.7-3ubuntu1
Python-2.6, 2.7
python-indicate: 0.6.1-0ubuntu1
Python-2.7
unity: 4.24.0-0ubuntu2b1
unity-2d: 4.12.0-0ubuntu1
telepathy-mission-control-5: 1:5.9.1-0ubuntu2
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Attached complete
INFO: Checking unread messages
INFO: Checking online status changing users
INFO: User anoxaemia is online
INFO: Found avatar for anoxaemia
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-anoxaemia.jpg
INFO: User olikuschel is online
INFO: Found avatar for olikuschel
INFO: User opampop is online
INFO: Found avatar for opampop
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-opampop.jpg
INFO: User deejaysevero is online
INFO: Found avatar for deejaysevero
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-deejaysevero.jpg
INFO: User tonfilm is online
INFO: Found avatar for tonfilm
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-tonfilm.jpg
INFO: User death.to.goemb is online
INFO: Found avatar for death.to.goemb
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-death.to.goemb.jpg
INFO: User jvulliet is online
INFO: Found avatar for jvulliet
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-jvulliet.jpg
INFO: User catweasel8 is online
INFO: Found avatar for catweasel8
INFO: User giorgio.rinolfi is online
INFO: Found avatar for giorgio.rinolfi
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-giorgio.rinolfi.jpg
INFO: User antokhio is online
INFO: Found avatar for antokhio
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-antokhio.jpg
INFO: User margithelin is online
INFO: Found avatar for margithelin
INFO: User solanina_ is online
INFO: Found avatar for solanina_
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-solanina_.jpg
INFO: Checking online presence
INFO: Checking file transfers
INFO: Check if Skype instance is running
INFO: Checking online presence
INFO: Check if Skype instance is running
INFO: Checking online presence
INFO: Check if Skype instance is running
etc..

**Comment by sagetarian** [comment]
nah, just start skype with the latest skype-wrapper :) with the dbpath commandline in your dconf settings .. the latest one is available on the PPA (0.5.4.2) 

i have provided what i think might be a fix

**Comment by bilderbuchi** [comment]
using 0ubuntu5-4.2~oneiric2 I get in the command line:
$ skype-wrapper 
Starting skype-wrapper
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Starting Skype
Corrupt JPEG data: 291 extraneous bytes before marker 0xd9

but no notifications. also on quitting skype, skype starts again.
I had the command in dconf-editor without the  --enable-dbus --use-session-dbus, right?

btw, I've seen that you provide a p4skype(?)package in the PPA, but I have python-skype  1.0.31.0-2 manually installed (as per the original instructions) - can that be relevant?

btw, the corrupt jpeg thing is a skype thing i think - i also got it yesterday when starting skype alone, without skype-wrapper

**Comment by sagetarian** [comment]
so wait .. is that all the output you get?? :? no "attaching to skype messages"?

**Comment by bilderbuchi** [comment]
nope, that's everything. I'll do a logout/in in ~10mins, and try with a fresh session to confirm, and will get back to you.

**Comment by bilderbuchi** [comment]
yes, I can confirm. after logout/in, starting skype-wrapper in terminal, and waiting a while until skype is online, terminal output is:
$ skype-wrapper 
Starting skype-wrapper
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Starting Skype
Corrupt JPEG data: 291 extraneous bytes before marker 0xd9

and log file (debug level verbose/0):
python-skype: 1.0.31.0-2
python-imaging: 1.1.7-3ubuntu1
Python-2.6, 2.7
python-indicate: 0.6.1-0ubuntu1
Python-2.7
unity: 4.24.0-0ubuntu2b1
unity-2d: 4.12.0-0ubuntu1
telepathy-mission-control-5: 1:5.9.1-0ubuntu2
INFO: Initializing Skype API
INFO: Starting Skype

If I quit skype it's immediately started again.

**Comment by bilderbuchi** [comment]
If I then kill everything, start skype (with dbpath) in a terminal
$ skype --dbpath=/media/windata/sharedfiles/Skype-linux
Corrupt JPEG data: 291 extraneous bytes before marker 0xd9
, then skype-wrapper in another terminal, I get
$ skype-wrapper 
Starting skype-wrapper
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Attached complete
INFO: Checking unread messages
WARNING: Checking unread messages failed: Skype command timeout
INFO: Checking online status changing users
INFO: User anoxaemia is online
INFO: Found avatar for anoxaemia
INFO: Wrote avatar to file /home/bilderbuchi/.thumbnails/normal/skype-wrapper-anoxaemia.jpg
...
INFO: Checking online presence
INFO: Checking file transfers
INFO: Check if Skype instance is running
INFO: Checking online presence
INFO: Check if Skype instance is running
INFO: Checking online presence
...

Notifications work, and there's only one icon in UMM. Everything seems fine as far as I can see.

**Comment by sagetarian** [comment]
awesome .. the latest version in the PPA should sort out the issues here

**Comment by bilderbuchi** [comment]
it's still in the build queue on LP. I'll try as soon as I get it (tomorrow probably)

**Comment by sagetarian** [comment]
strange .. LP says its done over here? here's a link to the deb if you need

https://github.com/sagetarian/skype-wrapper/blob/master/skype-wrapper_0ubuntu5-4.3_all.deb?raw=true

**Comment by bilderbuchi** [comment]
I think we can close this now, see #17.
